src/api/app/controllers/webui/request_controller.rb

```diff
@@ -41,25 +41,17 @@ def create
 
   def add_reviewer
     begin
-      opts = {}
-      case params[:review_type]
-      when 'review-user'
-        opts[:by_user] = params[:review_user]
-      when 'review-group'
-        opts[:by_group] = params[:review_group]
-      when 'review-project'
-        opts[:by_project] = params[:review_project]
-      when 'review-package'
-        opts[:by_project] = params[:review_project]
-        opts[:by_package] = params[:review_package]
-      end
-      opts[:comment] = params[:review_comment] if params[:review_comment]
-
       request = BsRequest.find_by_number!(params[:number])
-      request.addreview(opts)
+    rescue ActiveRecord::RecordNotFound => e
+      flash[:error] = "Unable to add review to request with id '#{params[:number]}': #{e.message}"
+      return
```

Comment:
If somehow `ActiveRecord::RecordNotFound` is raised and caught in this `rescue` block, this `return` will cause the UI to simply not respond. This is quite confusing, see here:

![nothing-happening](https://user-images.githubusercontent.com/1102934/181030406-305671a0-7c90-4344-a966-50c54593e935.gif)

The solution is either to duplicate the `redirect_to` call on line 57 below or to reorganize the code.

Reply:
Ops, true. On it.

Reply:
Given that we are rescuing only `ActiveRecord::RecordNotFound` exceptions, I changed the code from a `find_by_number!` to a `find_by_number`, and later checked with `.nil?`.